File: frontend/src/api/booking.js
import axios from 'axios';

const API_URL = "http://localhost:5000/api";
export const getBooking = async () => {
    try {
        const response = await axios.het('${API_URL}/booking');
        return response.data;
    }
    catch (error) {
        console.error("Error fetching bookings", error);
        throw error;
    }
    
};

export const createBooking = async (user_id, parking_spot_id, start_time, end_time) => {
    try {
        const response = await axios.post('${API_URL}/bookings', {
            user_id,
            parking_spot_id,
            start_time,
            end_time
        });
        return response.data;
    }
    catch (error) {
        console.error("Error creating booking", error);
        throw error;
    }
};

File: backend/Booking.cs
using System;

namespace ParkingApp;

public class Booking
{
    public int Id { get; set; }
    public int ParkingSpotId { get; set; } 
    public DateTime StartTime { get; set; }
    public DateTime EndTime { get; set; }

    public ParkingSpot? ParkingSpot { get; set; } 
}
